src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -227,9 +227,31 @@ private async Task MonitorServersAsync()
                     var eventArgs = await _serverDescriptionChangedQueue.DequeueAsync(monitorServersCancellationToken).ConfigureAwait(false); // TODO: add timeout and cancellationToken to DequeueAsync
                     ProcessServerDescriptionChanged(eventArgs);
                 }
-                catch
+                catch (OperationCanceledException) when (monitorServersCancellationToken.IsCancellationRequested)
                 {
-                    // TODO: log this somewhere...
+                    // ignore OperationCanceledException when monitor servers cancellation is requested
+                }
+                catch (Exception unexpectedException)
+                {
+                    // if we catch an exception here it's because of a bug in the driver
+
+                    var handler = _sdamInformationEventHandler;
+                    if (handler != null)
+                    {
+                        try
+                        {
+                            handler.Invoke(new SdamInformationEvent(() =>
+                                string.Format(
+                                    "Unexpected exception in MultiServerCluster.MonitorServersAsync: {0}",
+                                    unexpectedException.ToString())));
```

Comment:
Does ToString have enough information to find a bug?  LIke the stack trace?  If not, can we include more?

Reply:
Yes, ToString includes the full stack trace.